src/main/java/com/gameshop/ecommerce/security/auth/service/AuthenticationService.java

```diff
@@ -5,19 +5,81 @@
 import com.gameshop.ecommerce.utils.exception.JwtTokenException;
 import com.gameshop.ecommerce.web.user.model.User;
 import com.gameshop.ecommerce.web.user.service.UserService;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.authentication.AuthenticationManager;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.Optional;
+
 @Service
 @RequiredArgsConstructor
+@Slf4j
 public class AuthenticationService {
     private final AuthenticationManager authenticationManager;
     private final JwtService jwtService;
     private final UserService userService;
     private final PasswordEncoder passwordEncoder;
+    @Value("${spring.security.oauth2.client.registration.google.client-id}")
+    private String googleClientId;
+
+    public LoginResponse authenticateGoogle(String googleToken) {
+        Optional<User> verifiedUser = getUserFromGoogleIdToken(googleToken);
+        if (verifiedUser.isEmpty()) {
+            throw new JwtTokenException("Invalid Google token");
+        }
+        String accessToken = jwtService.generateAccessToken(verifiedUser.get());
+        String refreshToken = jwtService.generateRefreshToken(verifiedUser.get());
+        return new LoginResponse(accessToken, refreshToken);
+    }
+
+    private Optional<User> getUserFromGoogleIdToken(String token) {
+        try {
+            final var googleIdToken = new GoogleIdTokenVerifier.Builder(
+                    new NetHttpTransport(), new GsonFactory())
+                    .setAudience(List.of(googleClientId))
+                    .build()
+                    .verify(token);
+
+            if (googleIdToken != null) {
+                final var payload = googleIdToken.getPayload();
+                // check if user with this email already exists
+                User user = (User) userService.userDetailsService().loadUserByUsername(payload.getEmail());
```

Comment:
loadUserByUsername throws an exception if there is no user with the email in local db